router_v1/GluexRouter.sol

```diff
@@ -152,104 +161,98 @@ contract GluexRouter is EthReceiver {
      * @param desc The route description containing input, output, and fee details.
      * @param interactions The interactions encoded for execution by the executor.
      * @return finalOutputAmount The final amount of output token received.
+     * @return surplus The surplus amount on this route.
      * @dev Ensures strict validation of slippage, routing fees, and input/output parameters.
      */
     function swap(
         IExecutor executor,
         RouteDescription calldata desc,
         Interaction[] calldata interactions
-    ) external payable returns (uint256 finalOutputAmount) {
+    ) external payable returns (uint256 finalOutputAmount, uint256 surplus, uint256 slippage) {
 
-        // Validate routing fee
-        if (desc.routingFee > (desc.outputAmount * _MAX_FEE) / 10000) revert("Routing fee too high");
-        if (desc.routingFee < (desc.outputAmount * _MIN_FEE) / 10000) revert("Routing fee too low");
-
-        // Validate non-zero addresses
-        checkZeroAddress(desc.inputReceiver);
-        checkZeroAddress(desc.outputReceiver);
+        // Validate the route description
+        validateSwap(desc);
 
-        // Validate route parameters
-        if (desc.minOutputAmount <= 0) revert("Negative slippage limit");
-        if (desc.minOutputAmount > desc.outputAmount) revert("Slippage limit too large");
-        if (
-            desc.outputAmount <
-            desc.effectiveOutputAmount +
-                desc.partnerFee +
-                desc.routingFee -
-                _TOLERANCE ||
-            desc.outputAmount >
-            desc.effectiveOutputAmount +
-                desc.partnerFee +
-                desc.routingFee +
-                _TOLERANCE
-        ) revert("Invalid amounts");
-
-        // Handle native token input validation
-        if (address(desc.inputToken) == _nativeToken) {
-            if (msg.value != desc.inputAmount) revert("Invalid native token input amount");
-        } else {
-            if (msg.value != 0) revert("Invalid native token input amount");
-            desc.inputToken.safeTransferFromUniversal(
-                msg.sender,
-                desc.inputReceiver,
-                desc.inputAmount,
-                desc.isPermit2
-            );
-        }
+        // Transfer input tokens to the contract
+        desc.inputToken.safeTransferFromUniversal(
```

Comment:
This will attempt to transfer also when the input token is the native token. I believe this should only be triggered if inputToken != nativeToken. Same as it was previously implemented. Please correct this change.